pkg/testutils/testutils.go

```diff
@@ -208,3 +211,42 @@ func EnsureServiceHealthy(t testing.TB, grpcAddr, httpAddr string, transportCred
 		require.Equal(t, http.StatusOK, resp.StatusCode, "unexpected status code received from server")
 	}
 }
+
+// MustDefaultConfig returns default server config with the playground, tracing and metrics turned off.
+func MustDefaultConfig() *config.Config {
+	config := config.DefaultConfig()
+
+	config.Playground.Enabled = false
+	config.Metrics.Enabled = false
+
+	return config
+}
+
+// MustDefaultConfigWithRandomPorts returns default server config but with random ports for the grpc and http addresses
+// and with the playground, tracing and metrics turned off.
+// This function may panic if somehow a random port cannot be chosen.
+func MustDefaultConfigWithRandomPorts() *config.Config {
+	config := MustDefaultConfig()
+
+	httpPort, httpPortReleaser := TCPRandomPort()
+	defer httpPortReleaser()
+	grpcPort, grpcPortReleaser := TCPRandomPort()
+	defer grpcPortReleaser()
+
+	config.GRPC.Addr = fmt.Sprintf("0.0.0.0:%d", grpcPort)
+	config.HTTP.Addr = fmt.Sprintf("0.0.0.0:%d", httpPort)
+
+	return config
+}
+
+// TCPRandomPort tries to find a random TCP Port. If it can't find one, it panics. Else, it returns the port and a function that releases the port.
+// It is the responsibility of the caller to call the release function.
+func TCPRandomPort() (int, func()) {
+	l, err := net.Listen("tcp", "")
```

Comment:
This is where the port is bound by the OS. Any code after this should that attempts to bind to that port will fail.

Reply:
Correct, this is why it returns the releaser function, so that the code that asked for the port can actually use it.